File: a_easy/16_arith_geo/arith_geo.rb
def ArithGeo(arr)
  arith, geo = [], []

  [*(0..arr.length-2)].each { |n|
    arith << arr[n+1] - arr[n]
    geo << arr[n+1] / arr[n]
  }

  if arith.uniq.length == 1
    'Arithmetic'
  elsif geo.uniq.length == 1
    'Geometric'
  else
    -1
  end
end

# keep this function call here
puts ArithGeo(STDIN.gets)

File: a_easy/16_arith_geo/arith_geo.js
function arithGeo(arr) {
  var arith = [];
  var geo = [];

  for (i = 0; i < arr.length - 1; i++) {
    arith.push(arr[i + 1] - arr[i]);
    geo.push(arr[i + 1] / arr[i]);
  }

  if (arith.filter(pickUniqueValues).length === 1) {
    return "Arithmetic";
  } else if (geo.filter(pickUniqueValues).length === 1) {
    return "Geometric";
  } else {
    return -1;
  }
}

function pickUniqueValues(value, index, self) {
  return self.indexOf(value) === index;
}
